pkg/api/common/helper.go

```diff
@@ -337,3 +340,268 @@ func WrapAsParameter(s string) string {
 func WrapAsVerbatim(s string) string {
 	return fmt.Sprintf("',%s,'", s)
 }
+
+// GetDockerConfig transforms the default docker config with overrides. Overrides may be nil.
+func GetDockerConfig(opts map[string]string, overrides []func(*DockerConfig) error) (string, error) {
+	config := GetDefaultDockerConfig()
+
+	for i := range overrides {
+		if err := overrides[i](&config); err != nil {
+			return "", err
+		}
+	}
+
+	dataDir, ok := opts[ContainerDataDirKey]
```

Comment:
If I understand correctly, this is the real change we want to conditionally apply. If so, it seems we can just have another docker daemon template, and choose one of the two templates based on the condition. No?

Reply:
This PR is intended make arbitrary configuration changes to containerd/docker easier in the future. That's another reason I opted to model everything as structs. We probably will want to do additional tuning in the future and I'd prefer defaulting/mutating structs to growth of template options.